packages/data-context/src/util/weightedChoice.ts

```diff
@@ -13,20 +13,22 @@ export type WeightedAlgorithm = {
  * @param values array of values to choose from
  */
 const weightedChoice = (weights: number[], values: any[]) => {
-  if (weights.length !== values.length) {
-    throw new Error('The length of the weights and values must be the same')
+  if (weights.length > 0 && values.length > 0 && weights.length !== values.length) {
+    throw new Error('The length of the weights and values must be the same and greater than zero')
   }
```

Comment:
This doesn't validate that the arrays are non-empty

```suggestion
  if (weights.length === 0 || values.length === 0 || weights.length !== values.length) {
    throw new Error('The `weights` and `values` parameters must be the same non-zero length')
  }
```

Reply:
Not sure where my brain was for this.  Will fix and add tests.